server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -623,7 +698,43 @@ public int compare(File o1, File o2)
           if (Ints.tryParse(segmentDir.getName()) == null) {
             continue;
           }
-
+          QueryableIndex queryableIndex = null;
+          try {
+            queryableIndex = IndexIO.loadIndex(segmentDir);
```

Comment:
creating https://github.com/druid-io/druid/issues/1677 to track this separately.